morphct/system.py

```diff
@@ -96,6 +97,10 @@ def chromophores(self):
         """Return the chromophores in the system."""
         return self._chromophores
 
+    def carrier_data(self):
+        """Return the carrier data for data inspecting purposes"""
+        return self.carrier_data
```

Comment:
Because carrier data is already an attribute, I think this might cause an error to have a function with the same name.  If you look at the above example with the `chromophores` function, you can see that it has the `@property` decorator and returns the `_chromophores` attribute (note the leading underscore). The leading underscore hints that this is a private attribute and users should try to set it manually. [This stackoverflow post](https://stackoverflow.com/questions/14594120/python-read-only-property) gives some good examples of these "read-only" properties.

Reply:
[This article](https://www.programiz.com/python-programming/property) may also help clear up the property decorator.

Reply:
Ok I think I understood what you meant. the idea was correct but the syntax wasn't ideal? I pushed these changes